rules/integrations/azure/persistence_azure_service_principal_credentials_added.toml

```diff
@@ -0,0 +1,84 @@
+[metadata]
+creation_date = "2021/05/05"
+integration = ["azure"]
+maturity = "production"
+updated_date = "2025/03/26"
+
+[rule]
+author = ["Elastic", "Austin Songer"]
+description = """
+Identifies when new Service Principal credentials have been added in Microsoft Entra ID. In most organizations,
+credentials will be added to service principals infrequently. Hijacking an application (by adding a rogue secret or
+certificate) with granted permissions will allow the attacker to access data that is normally protected by MFA
+requirements.
+"""
+false_positives = [
+    """
+    Service principal credential additions may be done by a system or network administrator. Verify whether the
+    username, hostname, and/or resource name should be making changes in your environment. Credential additions from
+    unfamiliar users or hosts should be investigated. If known behavior is causing false positives, it can be exempted
+    from the rule.
+    """,
+]
+from = "now-9m"
+index = ["filebeat-*", "logs-azure*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Microsoft Entra ID Service Principal Credentials Added by Rare User"
+note = ""
```

Comment:
Did you intentionally remove the investigation guide?

Reply:
Unintentional, but will add an updated investigation guide.

Reply:
@Mikaayenson - investigation guide added, will resolve this when you have signed off.